docs: Improve dynatrace documentation (#708)
kyma-project/telemetry-manager: docs/user/integration/dynatrace/README.md

- Kyma as the target deployment environment
- The [Telemetry module](https://kyma-project.io/#/telemetry-manager/user/README) is [enabled](https://kyma-project.io/#/02-get-started/01-quick-install)
- Active Dynatrace environment with permissions to create new access tokens

> - - A Secret in the respective namespace in the Kyma cluster, holding the access tokens and endpoints for the Dynatrace instance. In this guide, the Secret is named `dynakube` and the namespace `dynatrace` as illustrated in this [example](https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/integration/dynatrace/secret-example.yaml).

Reviewer: ```suggestion
- A Secret in the respective namespace in the Kyma cluster, holding the access tokens and endpoints for the Dynatrace instance. In this guide, the Secret is named `dynakube` and the namespace `dynatrace` as illustrated in this [example](https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/integration/dynatrace/secret-example.yaml).
```

- Kyma as the target deployment environment
- The [Telemetry module](https://kyma-project.io/#/telemetry-manager/user/README) is [enabled](https://kyma-project.io/#/02-get-started/01-quick-install)
- Active Dynatrace environment with permissions to create new access tokens

> - A Secret in the respective namespace in the Kyma cluster, holding the access tokens and endpoints for the Dynatrace instance. In this guide, the Secret is named `dynakube` and the namespace `dynatrace` as illustrated in this [example](https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/integration/dynatrace/secret-example.yaml).

Reviewer 2: ```suggestion
- A Secret in the respective namespace in the Kyma cluster, holding the access tokens and endpoints for the Dynatrace instance. If you already have a Secret, you must know its name and namespace. If you don't have a Secret yet, this guide helps you [create it](#create-access-token-and-secret).
```

>   >**NOTE:** If you don't have the required access tokens, you can create the Secret [later](#create-access-token-and-secret).

Reviewer 2: I added this above
```suggestion
```

- The [Telemetry module](https://kyma-project.io/#/telemetry-manager/user/README) is [enabled](https://kyma-project.io/#/02-get-started/01-quick-install)
- Active Dynatrace environment with permissions to create new access tokens
- A Secret in the respective namespace in the Kyma cluster, holding the access tokens and endpoints for the Dynatrace instance. In this guide, the Secret is named `dynakube` and the namespace `dynatrace` as illustrated in this [example](https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/integration/dynatrace/secret-example.yaml).

>   >**NOTE:** If you don't have the required access tokens you can create the Secret [later](#create-access-token-and-secret).

Reviewer 2: ```suggestion
  >**NOTE:** If you don't have the required access tokens, you can create the Secret [later](#create-access-token-and-secret).
```

Reviewer 2: But really, if creating the Secret is part of this guide, why keep it as prerequisite at all?

Author: If the user already has access token he can create under pre-requisites. If he does not know how to create access tokens or does not know the scopes then he can create it later.

Reviewer 2: I'm not sure I understand. Purpose of the guide is to instruct them how to create access tokens, isn't it? So the assumption is that they don't have them yet. Otherwise half of the guide is not required. But maybe I misunderstand the purpose and our assumptions for this guide.

1. Install Dynatrace with the namespace you prepared earlier.
   >**NOTE:** By default, Dynatrace uses the classic full-stack injection. However, for better stability, we recommend using the [cloud-native fullstack injection](https://docs.dynatrace.com/docs/setup-and-configuration/setup-on-k8s/installation/cloud-native-fullstack).

> 2. In the DynaKube resource, configure the correct `apiUrl` of your environment.

Reviewer 3: ```suggestion
2. In the DynaKube resource, configure the correct `apiurl` of your environment.
```

Next, you set up the ingestion of custom span and Istio span data, and, optionally, custom metrics based on OTLP.

### Create Secret
1. To push custom metrics and spans to Dynatrace, set up a [dataIngestToken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens).

   Follow the instructions in [Dynatrace: Generate an access token](https://docs.dynatrace.com/docs/manage/access-control/access-tokens#create-api-token) and select the following scopes:

   - **Ingest metrics**
   - **Ingest OpenTelemetry traces**

> 2. Create an [apitoken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens) by selecting the template `Kubernetes: Dynatrace Operator`.

Reviewer 3: Write `apiToken` consistently with the Secret below

> 3. To create a new Secret containing your access tokens, replace the `{API_TOKEN}` and `{DATA_INGEST_TOKEN}` placeholder with the `apitoken` and `dataIngestToken` you created, replace the `{API_URL}` placeholder with the Dynatrace endpoint, and run the following command:

Reviewer 3: Same here

> 3. To create a new Secret containing your access tokens, replace the `{API_TOKEN}` and `{DATA_INGEST_TOKEN}` placeholder with the `apitoken` and `dataIngestToken` you created, replace the `{API_URL}` placeholder with the Dynatrace endpoint, and run the following command:

Reviewer 3: The placeholders have different names in the Secret example and the kubectl example below

   ```bash
   kubectl -n $DYNATRACE_NS create secret generic dynakube --from-literal="apiToken=<API_TOKEN>" --from-literal="dataIngestToken=<DATA_INGEST_TOKEN>" --from-literal="apiurl=<API_URL>"
   ```

>    verify the Secret created above looks similar to the [example Secret](https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/integration/dynatrace/secret-example.yaml)

Reviewer 2: ```suggestion
4. Verify the Secret you created looks similar to the [example Secret](https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/integration/dynatrace/secret-example.yaml).
```

Next, you set up the ingestion of custom span and Istio span data, and, optionally, custom metrics based on OTLP.

### Create Access Token and Secret

> To push custom metrics and spans to Dynatrace, set up [dataIngestToken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens).

Reviewer 2: Why not format instructions as numbered steps, just like in the previous section? 
```suggestion
1. To push custom metrics and spans to Dynatrace, set up a [dataIngestToken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens).
```

> Follow the instructions in [Dynatrace: Generate an access token](https://docs.dynatrace.com/docs/manage/access-control/access-tokens#create-api-token) and select the following scopes:

Reviewer 2: Within a numbered list, we'll need 3 blank spaces for indentation.
```suggestion
   Follow the instructions in [Dynatrace: Generate an access token](https://docs.dynatrace.com/docs/manage/access-control/access-tokens#create-api-token) and select the following scopes:
```

> - **Ingest metrics**

Reviewer 2: ```suggestion
   - **Ingest metrics**
```

> - **Ingest OpenTelemetry traces**

Reviewer 2: ```suggestion
   - **Ingest OpenTelemetry traces**
```

Additionally create [apitoken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens) by selecting template `Kubernetes: Dynatrace Operator`.

> To create a new Secret containing your access tokens, replace the `{API_TOKEN}` and `{DATA_INGEST_TOKEN}` placeholder with the `apitoken` and `dataIngestToken` you created, replace the `{API_URL}` placeholder with the Dynatrace endpoint, and run the following command:

Reviewer 2: ```suggestion
3. To create a new Secret containing your access tokens, replace the `{API_TOKEN}` and `{DATA_INGEST_TOKEN}` placeholder with the `apitoken` and `dataIngestToken` you created, replace the `{API_URL}` placeholder with the Dynatrace endpoint, and run the following command:
```

> ```bash

Reviewer 2: ```suggestion
   ```bash
```

> kubectl -n $DYNATRACE_NS create secret generic dynakube --from-literal="apiToken=<API_TOKEN>" --from-literal="dataIngestToken=<DATA_INGEST_TOKEN>" --from-literal="apiurl=<API_URL>"

Reviewer 2: ```suggestion
   kubectl -n $DYNATRACE_NS create secret generic dynakube --from-literal="apiToken=<API_TOKEN>" --from-literal="dataIngestToken=<DATA_INGEST_TOKEN>" --from-literal="apiurl=<API_URL>"
```

> ```

Reviewer 2: ```suggestion
   ```
```

Next, you set up the ingestion of custom span and Istio span data, and, optionally, custom metrics based on OTLP.

> ### Create Access Token and Secret

Reviewer 2: Maybe even:

```suggestion
### Create Secret
```

> >**NOTE:** if you have already created `dynakube` Secret in prerequisites then skip this step.

Reviewer 2: ```suggestion
>**NOTE:** If you already have a Secret, skip this section and proceed with [Ingest Traces](#ingest_traces).
You must know know the name and namespace of the Secret. As an [example](https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/integration/dynatrace/secret-example.yaml), this guide uses a Secret named `dynakube` in the namespace `dynatrace`.
```

1. To push custom metrics and spans to Dynatrace, set up a [dataIngestToken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens).

>    Follow the instructions in [Dynatrace: Generate an access token](https://docs.dynatrace.com/docs/manage/access-control/access-tokens#create-api-token) and select the following scopes:

Reviewer 2: are those scopes or templates (like in line 96)?

Author: both :), template is bunch of scopes and are pre-defined in dynatrace. There is no template defined for dataIngestToken so we have to mention specific scopes.

### Create Access Token

> 

Author: Added a note here with assumption that user might have the access tokens. If he does not know what or how to create access tokens and with which scopes he would create it later and we link it to the section.

> 

Author: May be I can add note here to skip it if already done in prerequeisites.

> 

Reviewer: I think that is not sufficient. Please will still continue executing the instructions one-by-one. We need something more clear in the chapter, please discuss with @NHingerl

> 

Reviewer 2: Question is, do we even want to keep the prerequisite or simply drop it? 

Or, in the prerequisites, we could state, "if you have an existing Secret you want to use, you need to know its xyz" (probably the name?).

And in the step in line 98, "If there's no Secret yet, create one now. In the following command, replace..."

> 

Author: I wanted to give an example of the secret and explain various contents of the secret. I thought to make it similar to Cloud Logging: https://github.com/kyma-project/telemetry-manager/blob/main/docs/user/integration/sap-cloud-logging/README.md#prerequisites. 

May be we can add an example after we create the secret ?

To push custom metrics and spans to Dynatrace, set up [dataIngestToken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens).

Follow the instructions in [Dynatrace: Generate an access token](https://docs.dynatrace.com/docs/manage/access-control/access-tokens#create-api-token) and select the following scopes:

- **Ingest metrics**
- **Ingest OpenTelemetry traces**

Additionally create [apitoken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens) by selecting template `Kubernetes: Dynatrace Operator`.
### Create Secret

> To create a new Secret containing your access tokens, replace the `{API_TOKEN}` and `{DATA_INGEST_TOKEN}` placeholder with the `apitoken` and `dataIngestToken` you created, replace the `{API_URL}` placeholder with the Dynatrace SaaS endpoint, and run the following command:

Reviewer: ```suggestion
To create a new Secret containing your access tokens, replace the `{API_TOKEN}` and `{DATA_INGEST_TOKEN}` placeholder with the `apitoken` and `dataIngestToken` you created, replace the `{API_URL}` placeholder with the Dynatrace endpoint, and run the following command:
```

### Create Secret

> To create a new Secret containing your access token, replace the `{API_TOKEN}` placeholder with the token you created, replace the `{API_URL}` placeholder with the Dynatrace SaaS endpoint and run the following command:

Reviewer 2: ```suggestion
To create a new Secret containing your access token, replace the `{API_TOKEN}` placeholder with the token you created, replace the `{API_URL}` placeholder with the Dynatrace SaaS endpoint, and run the following command:
```

```bash

> kubectl -n $DYNATRACE_NS create secret generic dynakube --from-literal="apiToken=<API_TOKEN>" --from-literal="apiurl=<API_URL>"

Reviewer: What's with the dataIngestToken? That needs to be used in our pipelines, isn't it?

Author: added the dataIngestToken

> kubectl -n $DYNATRACE_NS create secret generic dynakube --from-literal="apiToken=<API_TOKEN>" --from-literal="apiurl=<API_URL>"

Reviewer: What's with the dataIngestToken? That needs to be used in our pipelines, isn't it?

- **Ingest OpenTelemetry traces**

> Additionally create [apitoken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens) by selecting template `Kubernetes: Dynatrace Operator`.

Reviewer 2: ```suggestion
2. Create an [apitoken](https://docs.dynatrace.com/docs/manage/access-control/access-tokens) by selecting the template `Kubernetes: Dynatrace Operator`.
```

    The default configuration has the **randomSamplingPercentage** property set to `1.0`, meaning it samples 1% of all requests. To change the sampling rate, adjust the property to the desired value, up to 100 percent.

    > **CAUTION:** Be cautious when you configure the **randomSamplingPercentage**:

>     > - Traces might consume a significant storage volume in Dynatrace.

Reviewer 3: Is that relevant? Maybe rather refer to Dynatrace pricing. Check if there is a price per span.

        output:
            otlp:
                endpoint:
                    valueFrom:
                        secretKeyRef:
                            name: dynakube
                            namespace: ${DYNATRACE_NS}
                            key: apiurl

>                 path: v2/otlp/v1/metrics

Reviewer 3: This path seems to be for metrics, not traces

Author: fixed

        output:
            otlp:
                endpoint:
                    valueFrom:
                        secretKeyRef:
                            name: dynakube
                            namespace: ${DYNATRACE_NS}
                            key: apiurl
                path: v2/otlp/v1/metrics
                headers:
                    - name: Authorization
                      prefix: Api-Token
                      valueFrom:
                          secretKeyRef:
                              name: dynakube
                              namespace: ${DYNATRACE_NS}

>                               key: apiToken

Reviewer: The apiToken is the one used by the oneAgent usually and does not cover the required scopes

        output:
            otlp:
                endpoint:

>                     valueFrom:

Reviewer 3: Add path

Author: fixed

        output:
            otlp:
                endpoint:
                    valueFrom:
                        secretKeyRef:
                            name: dynakube
                            namespace: ${DYNATRACE_NS}
                            key: apiurl

>                             path: v2/otlp/v1/metrics

Reviewer 3: Wrong indentation
